component/identity/src/main/java/org/gatein/portal/idm/impl/store/attribute/ExtendedAttributeManager.java

```diff
@@ -44,24 +45,33 @@ public ExtendedAttributeManager(IdentitySessionImpl session) {
 
   @Override
   public boolean validatePassword(User user, String password) throws IdentityException {
-
+    Attribute salt = getAttribute(user.getKey(), OLD_PASSWORD_SALT_USER_ATTRIBUTE);
     Attribute salt128 = getAttribute(user.getKey(), PASSWORD_SALT_USER_ATTRIBUTE);
-    if (salt128 != null) {
-      return super.validatePassword(user, password);
-    } else {
-      DatabaseReadingSaltEncoder oldCredentialEncoder = new DatabaseReadingSaltEncoder();
-      oldCredentialEncoder.setIdentitySession(getIdentitySession());
-      oldCredentialEncoder.initialize(getCredentialEncoderProps(),
-                                      getIdmService().getIdentityConfiguration().getIdentityConfigurationRegistry());
+    if (salt != null && salt128 != null) {
+      return super.validatePassword(user,
```

Comment:
```suggestion
      //password has been updated by upgradePlugin, and user was not connected since
      //passwordHash was created like this passwordHash = newEncoder(oldEncoder(password))
      return super.validatePassword(user,
```